### DISCUSSION TITLE
Computed defaults for a ZodObject

### DISCUSSION THREAD

**Discussion by zayenz** [colinhacks/zod#2095] [Q&A] [answered]
Is there some way to make a computed default for a Zod object, and keeping the type as a ZodObject? If not, would that be a good idea or not?

I have several cases where I would like to specify a default for a field that is computed from the values of other fields. This can be done by using a transform, but unfortunately that leaves the resulting type as a ZodEffects, which limits what it can be used for.

My particular use-case is that I have several different base ZodObjects that are extended with a tag and then joined into a discriminated union. The extension and the discriminated union requires (AFAIK) that I work with ZodObjects which I get, and that precludes using the general transform operation.

My current solution is to do the computation of the default value for the missing fields at the use-site instead, but that feels more messy than it needs to be.

**Comment by JacobWeisenburger** [comment, marked as answer]
As far as I know, I don't think what you are asking for is possible. Sorry about that.
